### DISCUSSION TITLE
How to pass SqlServerResource host and port to projects and containers

### DISCUSSION THREAD

**Discussion by paulomorgado** [microsoft/aspire#6653] [Q&A]
I need to pass the host and port from a `SqlServerResource` to resources that reference it.

If the resource is a container, I need the values from the container network, while if it's a project resource, it needs the host host and port.

As an example, how to build a JDBC connection string?

How to do this transparently?

It works if it's a connection string.

**Comment by davidfowl** [comment]
See the discussion here https://github.com/dotnet/aspire/issues/2111

**Reply by paulomorgado** [reply]
So, when I thought about doing [this](https://github.com/dotnet/aspire/issues/2111#issuecomment-2350929098) and didn't even try because there must be a better way, I should have kept going?

Wasn't there a property that gave me the container host name so that I don't have to deal with `host.docker.internal` vs. `host.containers.internal`? The `AllocatedEndpoint` of `EndpointAnnotation`?

**Reply by davidfowl** [reply]
You don't need to deal with that, if you use the Host expression for the endpoint it'll resolve to the "right one" based on the resource consuming this information.

I updated the thread.
